contracts/multisig/src/key.rs

```diff
@@ -499,14 +532,31 @@ mod ed25519_tests {
     }
 
     #[test]
-    fn test_try_from_hexbinary_to_eccdsa_public_key_fails() {
+    fn test_try_from_hexbinary_to_ed25519_public_key_fails() {
         let hex = HexBinary::from_hex("049b").unwrap();
         assert_eq!(
             PublicKey::try_from((KeyType::Ed25519, hex.clone())).unwrap_err(),
             ContractError::InvalidPublicKey
         );
     }
 
+    #[test]
+    fn validate_ed25519_public_key_not_in_curve() {
+        // Valid compressed point format, which should produce an invalid point when decompressed
```

Comment:
why is it valid compressed but not decompressed?

Reply:
the compressed format is a single coordinate and ed25519 accepts any 32 bytes, doesn't really say it's a valid point or not on the curve, but once you substitute `y` in the curve equation, then `x` would get you an invalid number like square root of a negative number

Reply:
I don't think I get it. The decompressed format is completely derived from the compressed one without extra information. So how is it possible that the compressed version could be valid?

Reply:
The only thing valid is the format, ed25519 spec says any 32 bytes array is accepted, the crate doesn't really do any extra validation when creating a `CompressedEdwardsY`, but it will fail trying to decompress.